eng/pipelines/templates/steps/vcpkg.yml

```diff
@@ -17,46 +11,23 @@ steps:
       Write-Host "##vso[task.prependpath]$TargetPath"
       Write-Host "##vso[task.setvariable variable=VCPKG_INSTALLATION_ROOT]$TargetPath"
       Write-Host "##vso[task.setvariable variable=VcpkgCommit]$VcpkgCommit"
+      Write-Host "##vso[task.setvariable variable=VCPKG_BINARY_SOURCES_SECRET;issecret=true;]clear;x-azblob,https://cppvcpkgcache.blob.core.windows.net/public-vcpkg-container,,read"
     displayName: Set Vcpkg Variables
-    condition: >-
-      and(
-        succeeded(),
-        not(eq(variables['${{ parameters.DependenciesVariableName }}'], ''))
-      )
 
-  # This task times out after 10 minutes. An analysis of a subset of Vcpkg Cache
-  # runs showed that after 10 minutes the cache step is very unlikely to
-  # succeed.
-  - task: Cache@2
+  - task: PowerShell@2
     inputs:
-      key: >-
-        $(Agent.JobName)
-        | "$(VcpkgCommit)"
-        | $(Agent.Os)
-        | $(${{ parameters.DependenciesVariableName }})
-      path: $(VCPKG_INSTALLATION_ROOT)
-      cacheHitVar: VcpkgRestoredFromCache
-    displayName: Vcpkg Cache
-    timeoutInMinutes: 10
-    condition: >-
-      and(
-        succeeded(),
-        not(eq(variables['${{ parameters.DependenciesVariableName }}'], '')),
-        not(eq(variables['Skip.VcpkgCache'], 'true'))
-      )
+      pwsh: true
+      targetType: filePath
+      filePath: eng/scripts/Set-VcpkgWriteModeCache.ps1
+      arguments: -StorageAccountKey '$(cpp-vcpkg-cache-storage-key)'
+    displayName: Set Vcpkg Write-mode Cache
+    condition: eq(variables['System.TeamProject'], 'internal')
```

Comment:
We probably also want to add `condition: and(succeeded(), eq(variables['System.TeamProject'], 'internal'))` but that can be a later cleanup.

Reply:
https://github.com/Azure/azure-sdk-for-cpp/issues/3480